examples/apps/ai-collab/src/actions/task.ts

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+"use server";
+
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+import { createJsonTranslator, createOpenAILanguageModel, type Result } from "typechat";
+import { createTypeScriptJsonValidator } from "typechat/ts"; // eslint-disable-line import/no-internal-modules
+import { v4 as uuidv4 } from "uuid";
+
+import type { Task, TaskGroup } from "@/types/task";
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+// Todo: Make use of system prompts with the open ai model, this may yield better results.
+// I am unsure if this is possible using typechat as a controller of the underlying open AI API's
+
+export async function editTask(workItem: Task, specificAsk: string): Promise<Result<Task>> {
+	const OPEN_AI_KEY = process.env.OPEN_AI_KEY;
+	if (OPEN_AI_KEY === undefined) {
+		throw new Error("OPEN_AI_KEY environment variable is not set");
+	}
+	const model = createOpenAILanguageModel(OPEN_AI_KEY, "gpt-4o");
+	const typesFile = fs.readFileSync(path.join(__dirname, "../../src/types/task.ts"), "utf8");
+	const validator = createTypeScriptJsonValidator<Task>(typesFile, "Task");
+	const translator = createJsonTranslator(model, validator);
+
+	const prompt = `You are a manager that is helping out with a project management tool. You have been asked to edit a task. \n\n
+	The task is as follows:
+	\n\n \`\`\`
+	${JSON.stringify(workItem)}
+	\n\n \`\`\`
+
+	This is the specific ask you have been given: "${specificAsk}"
+	`;
```

Comment:
nit: seems like factoring this out into a function might make it easier to work with. Alternatively, a template of some kind, since escaping the backticks (and other stuff) is error prone.